packages/signer-ledger-ton/src/signer.ts

```diff
@@ -0,0 +1,186 @@
+
+      const response = await app.getAddress(hdPath, {
+        bounceable: this.config.bounceable,
+        chain,
+        testOnly
+      })
+
+      this.accounts.set(response.address.toLowerCase(), {
+        hdPath: account.hdPath,
+        publicKey: response.publicKey
+      })
+    })
+  }
+
+  /**
+   * Signs the provided data using the private key associated with the signer's address.
+   *
+   * @param signerAddress - The address of the signer
+   * @param signerData - The data to be signed, which can be a raw message or custom data
+   *
+   * @returns A promise that resolves to an object containing the signature and public key.
+   */
+  async sign (signerAddress: string, signerData: SignerData): Promise<{ sig: Signature; pk: Uint8Array }> {
+    if (this.app === undefined) {
+      throw new Error('LedgerTonSigner not initialized. Did you forget to call init()?')
+    }
+    this.logger.info(`signing data from address: ${signerAddress} with Ledger Ton App`)
+
+    if (signerData.data === undefined) {
+      throw new Error('missing message to sign')
+    }
+
+    const data: TonSigningData = signerData.data
+    const account = this.getAccount(signerAddress)
+    const path = account.hdPath
+      .replace('m/', '')
+      .split('/')
+      .map((i: string) => Number(i.replace("'", '')))
+
+    if (data.tx.messages !== undefined && data.tx.messages.length > 1) {
+      throw new Error(
+        'Ledger App can only sign one message per transaction. The requested TX has ' +
+          data.tx.messages.length +
+          ' messages'
+      )
+    }
+
+    const msg = data.tx.messages !== undefined ? data.tx.messages[0] : undefined
+    const timeout = data.txArgs.timeout || data.tx.validUntil || Math.floor(Date.now() / 1000) + 180 // 3 minutes
+
+    let payload: TonPayloadFormat | undefined = undefined
+    if (msg !== undefined) {
+      if (msg.payload !== undefined && !(msg.payload instanceof Cell)) {
+        throw new Error('Ledger App can only sign Cell payloads')
+      }
+      payload = {
+        type: 'unsafe',
+        message: msg.payload === undefined ? Cell.EMPTY : (msg.payload as Cell)
+      }
+    }
+
+    const common = {
+      sendMode: data.txArgs.sendMode,
+      seqno: data.txArgs.seqno,
+      timeout,
+      payload
+    }
+
+    const args =
+      msg !== undefined
+        ? {
+            to: Address.parse(msg.address),
+            bounce: msg.bounceable,
+            amount: msg.amount,
+            stateInit: msg.stateInit,
+            ...common
+          }
+        : {
+            // A transaction without message is still a valid TX (e.g. for deploying wallet contract) however,
+            // the Ledger App requires one message to be present in the transaction,
+            // therefore we create a dummy message with zero amount and no stateInit
+            // that sends 0 tokens to self
+            to: Address.parse(signerAddress),
+            bounce: false,
```

Comment:
```suggestion
            bounce: true,
```

This looks like a supplied default. IMHO bounce should default to `true`. If it default to `false` then the default usage can end up in lost funds in case of errors.

Reply:
What funds sir?, see one line below:
```
amount: BigInt(0),
```

Also no messages are included in that TX